src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +41,104 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading, file]);
+
+  const fileChangedHandler = (event: any) => {
+    setFileInputText(event.target.files[0].name);
+    setFile(event.target.files[0]);
+  };
+
+  const handleUploadClick = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (file) {
+      const formData = new FormData();
+      formData.append('file', file);
+      setUploading(true);
+      const response = await fetch(
+        `${process.env.REACT_APP_BACKEND_ADDRESS}/image`,
+        {
+          method: 'POST',
+          body: formData,
+        }
+      );
+      const body = await response.json();
+      setResponseMessage(body.message);
+      setUploading(false);
+      setFile('');
+      setFileInputText(DEFAULT_FILE_INPUT_TEXT);
+      setShowToast(true);
+    }
+  };
 
   return (
     <Container>
       <Row>
+        <Col>
+          <Form onSubmit={handleUploadClick}>
+            <div className="input-group">
+              <div className="input-group-prepend">
+                <Button
+                  variant="secondary"
+                  type="submit"
+                  disabled={file === '' || isUploading}
+                >
+                  {isUploading ? (
+                    <Spinner
+                      as="span"
+                      animation="border"
+                      size="sm"
+                      role="status"
+                      aria-hidden="true"
+                    />
+                  ) : (
+                    'Upload Image'
+                  )}
+                </Button>
+              </div>
+              <div className="custom-file">
+                <input
+                  type="file"
+                  onChange={fileChangedHandler}
+                  aria-describedby="inputGroupFileAddon01"
+                  accept="image/*"
+                  className="custom-file-input"
+                />
+                <label className="custom-file-label" htmlFor="inputGroupFile01">
+                  {fileInputText}
+                </label>
+              </div>
+            </div>
+          </Form>
+        </Col>
+      </Row>
+      <Row className="content">
         {hasError && <div>Error encountered while loading images.</div>}
         {stimuliUrls && generateCards(stimuliUrls)}
       </Row>
+      {showToast && (
+        <Row>
+          <Col>
+            <div>
+              <Toast
+                style={{
+                  left: '50%',
+                  bottom: '0%',
+                  position: 'fixed',
+                  zIndex: 10,
+                  transform: 'translate(-50%, -50%)',
+                }}
```

Comment:
```suggestion
                style={{
                  position: 'sticky',
                }}
```

Try this styling instead to achieve the behavior of sticking to the bottom of the page. This is how it's done on the Experiments page and the less inline CSS styling we have, the better.